src/evtx_parser.rs

```diff
@@ -155,16 +155,25 @@ impl<T: ReadSeek> EvtxParser<T> {
         self
     }
 
-    pub fn allocate_chunk(data: &mut T, chunk_number: u16, validate_checksum: bool) -> Result<EvtxChunkData, Error> {
+    pub fn allocate_chunk(data: &mut T, chunk_number: u16, validate_checksum: bool) -> Result<Option<EvtxChunkData>, Error> {
         let mut chunk_data = Vec::with_capacity(EVTX_CHUNK_SIZE);
-        data.seek(SeekFrom::Start(
-            (EVTX_FILE_HEADER_SIZE + chunk_number as usize * EVTX_CHUNK_SIZE) as u64,
-        ))?;
+        let chunk_offset = EVTX_FILE_HEADER_SIZE + chunk_number as usize * EVTX_CHUNK_SIZE;
 
-        data.take(EVTX_CHUNK_SIZE as u64)
+        data.seek(SeekFrom::Start(chunk_offset as u64))?;
+
+        let amount_read = data.take(EVTX_CHUNK_SIZE as u64)
             .read_to_end(&mut chunk_data)?;
 
-        EvtxChunkData::new(chunk_data, validate_checksum)
+        if amount_read !=  EVTX_CHUNK_SIZE {
+            return Err(format_err!("Reached EOF while trying to read a chunk"));
+        }
+
+        // There might be empty chunks in the middle of a dirty file.
+        if chunk_data.iter().all(|&x| x == 0) {
```

Comment:
I think it’s clearer to deref the x instead of using borrow on the lambda parameter.